glide-core/redis-rs/redis/tests/auth.rs

```diff
@@ -0,0 +1,310 @@
+                    .await?;
+                Ok(Connection::Standalone(connection))
+            }
+        }
+    }
+
+    fn get_connection_info(cluster: &TestContext, password: Option<String>) -> ConnectionInfo {
+        let addr = cluster.server.connection_info().addr.clone();
+        ConnectionInfo {
+            addr,
+            redis: RedisConnectionInfo {
+                password,
+                ..Default::default()
+            },
+        }
+    }
+
+    fn get_builder(cluster: &TestClusterContext, password: Option<String>) -> ClusterClientBuilder {
+        let mut builder = ClusterClientBuilder::new(cluster.nodes.clone());
+        if let Some(password) = password {
+            builder = builder.password(password);
+        }
+        builder
+    }
+
+    async fn set_password(password: &str, conn: &mut Connection) -> RedisResult<()> {
+        let mut set_auth_cmd = cmd("config");
+        set_auth_cmd.arg("set").arg("requirepass").arg(password);
+        match conn {
+            Connection::Cluster(cluster_conn) => cluster_conn
+                .route_command(&set_auth_cmd, ALL_SUCCESS_ROUTE)
+                .await
+                .map(|_| ()),
+            Connection::Standalone(standalone_conn) => set_auth_cmd
+                .query_async::<_, ()>(standalone_conn)
+                .await
+                .map(|_| ()),
+        }
+    }
+
+    async fn kill_connection(con: &mut ClusterConnection) {
+        let mut kill_cmd = cmd("client");
+        kill_cmd
+            .arg("kill")
+            .arg("type")
+            .arg("normal")
+            .arg("skipme")
+            .arg("yes");
+        con.route_command(&kill_cmd, ALL_SUCCESS_ROUTE)
+            .await
+            .unwrap();
+    }
+
+    async fn reset_connection_standalone(con: &mut MultiplexedConnection) {
+        let _: () = cmd("reset").query_async(con).await.unwrap();
+    }
+
+    #[tokio::test]
+    #[serial_test::serial]
+    async fn test_replace_password_cluster() {
+        let cluster_context = TestClusterContext::new(3, 0);
+
+        // Create a management connection to set the password
+        let mut management_connection =
+            match create_connection(None, ConnectionType::Cluster, Some(&cluster_context), None)
+                .await
+                .unwrap()
+            {
+                Connection::Cluster(conn) => conn,
+                _ => panic!("Expected ClusterConnection"),
+            };
+
+        // Set the password using the unified function
+        let mut management_conn = Connection::Cluster(management_connection.clone());
+        set_password(PASSWORD, &mut management_conn).await.unwrap();
+
+        // Test that we can't connect without password
+        let connection_should_fail =
+            create_connection(None, ConnectionType::Cluster, Some(&cluster_context), None).await;
+        assert!(connection_should_fail.is_err());
```

Comment:
can you check that it's an auth error

Reply:
Here it's io error, it's handling of the try connect which exist, I added handling to AUTH error, handling for connection is part of the already built client.
But the error message will include my error. we can enhance the connection error handling in another pr to filter the errors better.

Reply:
Ill add error message check, and in another place check for error type.